### PR TITLE
Cleanup defer from SetFeatureGateDuringTest function call

### PR DESCRIPTION
**Pull request by serathius**
Followup from https://github.com/kubernetes/kubernetes/pull/123732
/kind cleanup

```release-note
NONE
```
/assign @liggitt

### CODE REVIEW

**Comment by serathius** [comment]
Stupid Intelij, it reformated and reordered imports in all files I touched. Need to review file by file to make sure I didn't git added it by mistake.

**Comment by serathius** [comment]
Should be good now.

File: hack/verify-test-featuregates.sh [hunk 1]
```diff
@@ -36,19 +36,19 @@ if [[ -n "${direct_sets}" ]]; then
...
   echo "Use this invocation instead:" >&2
-  echo "  defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.<FeatureName>, <value>)()" >&2
+  echo "  featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.<FeatureName>, <value>)" >&2
   echo >&2
   rc=1
 fi
 
 # find test files calling SetFeatureGateDuringTest and not calling the result
-missing_defers=$(git grep "\\.SetFeatureGateDuringTest" -- '*_test.go' | grep -E -v "defer .*\\)\\(\\)$") || true
+missing_defers=$(git grep "\\.SetFeatureGateDuringTest" -- '*_test.go' | grep -E "defer .*\\)\\(\\)$") || true
```
Comment (liggitt) [inline hack/verify-test-featuregates.sh:45]: this whole block can be dropped, right? this is now ensuring you *don't* call this with defer (which you can't do anyway?) and the warning output "Always make a deferred call" is no longer correct

File: staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go [hunk 2]
```diff
@@ -37,28 +37,26 @@ func init() {
 // SetFeatureGateDuringTest sets the specified gate to the specified value for duration of the test.
 // Fails when it detects second call to the same flag or is unable to set or restore feature flag.
 // Returns empty cleanup function to maintain the old function signature that uses defer.
```
Comment (liggitt) [inline staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go:39]: drop this line of the godoc

**Review by liggitt — COMMENTED** [review]
a couple nits, needs a rebase, lgtm otherwise

File: staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go (hunk 2 above)
Comment (serathius) [reply staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go:39]: Done

**Comment by serathius** [comment]
Addressed both comments and rebased.

**Comment by serathius** [comment]
No lgtm :( 
@liggitt do we need anyone's else review?

**Comment by liggitt** [comment]
/lgtm

oops, got dropped trying to drop the needs-rebase label

**Pull request merged**